### ISSUE TITLE
HTTP Error 422 (using cask-repair)

### ISSUE THREAD

**Issue by giannitm** [vitorgalvao/tiny-scripts#64]
I've started having this HTTP 422 error after setting up 2FA on my GitHub account. I noticed that something similar has been occurring recently and was supposed to have been fixed with this commit: https://github.com/vitorgalvao/tiny-scripts/pull/62 

Here's the error message I get from `cask repair`:

Is everything correct? ([y]es / [n]o / [e]dit) y
Submitting…
Username for 'https://github.com': giannitm
Password for 'https://giannitm@github.com': 
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/giannitm/homebrew-drivers/'
Error creating pull request: Unprocessable Entity (HTTP 422)
Invalid value for "head"
Error creating pull request: Unprocessable Entity (HTTP 422)
Invalid value for "head"

There was an error submitting the pull request. Please open a bug report on the repo for this script.

**Comment by vitorgalvao** [comment]
I’ll be blunt: I’m pretty tired of debugging that specific issue. It is `hub`-related (search for the error on Google) and generally happens with certain git configs. I can’t test them all. You can see from the other issues (e.g. #20) how much time I’ve already wasted on it.

Unfortunately, I won’t keep shooting in the dark. I’m happy to take a PR that squashes the problem for good, but don’t intend to waste much longer trying to diagnose it. It only affects a small set of users and is no longer worth the commitment and frustration.

**Comment by phiggins** [comment]
Hi @vitorgalvao,

I recently ran into this issue, and I suspect it is due to me having 2FA on my github account and cask-repair attempting to use HTTPS instead of SSH. I was able to successfully use cask-repair after doing the following:

```
Petes-MBP:homebrew-cask pete$ cd Library/Taps/caskroom/homebrew-cask/
Petes-MBP:homebrew-cask pete$ git remote -vv
cask-repair	https://github.com/phiggins/homebrew-cask (fetch)
cask-repair	https://github.com/phiggins/homebrew-cask (push)
origin	https://github.com/caskroom/homebrew-cask (fetch)
origin	https://github.com/caskroom/homebrew-cask (push)
Petes-MBP:homebrew-cask pete$ git remote set-url cask-repair git@github.com:phiggins/homebrew-cask.git
Petes-MBP:homebrew-cask pete$ git remote -vv
cask-repair	git@github.com:phiggins/homebrew-cask.git (fetch)
cask-repair	git@github.com:phiggins/homebrew-cask.git (push)
origin	https://github.com/caskroom/homebrew-cask (fetch)
origin	https://github.com/caskroom/homebrew-cask (push)
```

After this I was able to open a PR: https://github.com/caskroom/homebrew-cask/pull/30517

Thanks for this script! 🏆

**Comment by vitorgalvao** [comment]
@phiggins That’s definitely not the problem. I have 2FA and use HTTPS.

**Comment by giannitm** [comment]
I have managed to rectify the problem by using one of my personal access tokens instead of my password, when prompted by the `cask-repair` script to enter my GitHub password. I tried this previously, but it didn't work at the time, however I was just able to submit a pull request using `cask-repair` via HTTPS, with 2FA enabled using this method.

Not sure why it worked this time round, but I'm just glad the issue is fixed for me.

Sorry for bugging you unnecessarily, @vitorgalvao.
